### PR TITLE
Set job launcher for NDMIS job to be a read only job launcher

### PR DESCRIPTION
**Pull request by awhodgson256**
Set job launcher for NDMIS job to be a read only job launcher referencing the read only job repository in the batch job's configuration

## What does this pull request do?

Change how NDMIS report is launched, to use the batch job launcher instead of read / write launcher

## What is the intent behind these changes?

Fix problem with NDMIS report to allow it to run against read replica, thereby imoroving performance,

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jobs/oneoff/OnStartupJobLauncherFactory.kt [hunk 1]
```diff
@@ -44,6 +45,22 @@ class OnStartupJobLauncherFactory(
       return exitCodeMapper.intValue(execution.exitStatus.exitCode)
     }
 
+    return makeLauncher(job.name, entryPoint)
+  }
+  fun makeReadOnlyBatchLauncher(job: Job, readOnlyJobLauncher: TaskExecutorJobLauncher): ApplicationRunner {
```
Comment (kth13) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jobs/oneoff/OnStartupJobLauncherFactory.kt:50]: makeBatchLauncher and makeReadOnlyBatchLauncher almost the same code with the only the launcher is difference. could you please create the readonlylauncher as another bean and as that as a parameter to a private method inside. That private method will be used both by `makeBatchLauncher` and makeReadOnlyBatchLauncher`

Comment (awhodgson256) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jobs/oneoff/OnStartupJobLauncherFactory.kt:50]: No worries Kumar, will work that up this morning.

Comment (awhodgson256) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsinterventionsservice/jobs/oneoff/OnStartupJobLauncherFactory.kt]: Refactored as discussed @kth13 please review

**Review by kth13 — APPROVED** [review]
LGTM

**Pull request merged**